### PR TITLE
Frontend: Copy to clipboard without HTTPS

### PR DESCRIPTION
**Pull request by haimkastner**
Allow copy content to the clipboard even if the "window.navigator.clipboard" doesn't support

### CODE REVIEW

File: frontend/src/common/util.js [hunk 1]
```diff
@@ -145,4 +145,43 @@ export default class Util {
     console.log(`${label}: ${now.getTime() - start.getTime()}ms`);
     start = now;
   }
+
+  static async copyToMachineClipboard(text) {
+    if(window.navigator.clipboard) {
+      await window.navigator.clipboard.writeText(text);
+    } else { 
+      // Clapboard is available only in HTTPS pages. see https://web.dev/async-clipboard/
```
Comment (alexislefebvre) [inline frontend/src/common/util.js:153]:
```suggestion
      // Clipboard is available only in HTTPS pages. see https://web.dev/async-clipboard/
```

**Comment by Fylax** [comment]
Out of curiosity: is there any particular reason to make a new `Util` method instead of going for a more classical `prototype` that prevents refactoring?

**Comment by haimkastner** [comment]
Hi @Fylax,

I didn't fully understand what is your suggestion.

Do you mean to implement a `writeText` for `window.navigator.clipboard` if it does not exist? 

If so, I don't think it's a good idea to change the browser objects, and also where do you think we should implement it? in the `app.js` file?

**Comment by Fylax** [comment]
Hi @haimkastner,
Indeed it is what I am asking.

I'm not stating is the correct solution, just wondering if it can come handy.

Usually I create a `Prototypes` module that I load in application initialization. Common examples are `string.replaceAll`.

This mechanism allows me to have a uniform API without having to put lot of logic to discriminate between different browser versions and/or HTTP vs HTTPS.

Why would you be agains implementing `writeText` inside `windows.navigator.clipboard`?

**Comment by haimkastner** [comment]
Hi @Fylax,

It's only my thought, but I think that it's cool to extend an object prototype with some extra methods like your example, extending `replace-all` for the string prototype.

But to **override** existing object behavior that you don't own and you don't have full control over it such as the browser `window` object can make issues that you can't expect to, and In addition, in a case of bug/issue in the logic of the extension, no one can guess it's a problem in the prototype extensions that implemented somewhere in the app initialization since it's an official browser API.

**Comment by Fylax** [comment]
Hi @haimkastner,

thank you for your clarification, which makes completely sense now.

For any mantainer stepping here, you can assume my comments as a review: so far, the code is good.

**Comment by lastzero** [comment]
Wouldn't browsers likewise prevent copying from a textarea? Surprises me...

**Comment by haimkastner** [comment]
Yes...
I tested it on Chrome version 90.0.4430.85 (Official Build) (64-bit) Windows 10.

**Comment by Fylax** [comment]
@lastzero the `document.execCommand` API does not have any limit among HTTP/HTTPS, so it is expected to work out of the box.

Thinking of it, the only suggestion I might suggest to @haimkastner is adding a check to that API existence to avoid calling it and just throw an error if it doesn't.

Something like
```js
if(window.navigator.clipboard) {
  await window.navigator.clipboard.writeText(text);
} else if (document.execCommand) {
  // your code here
} else {
  throw new Error('Failed copying to clipboard')
}
```

Any thoughts on that?

**Comment by haimkastner** [comment]
@Fylax that sounds great to me, I will implement it soon.

**Comment by lastzero** [comment]
Thanks! We'll wait with merging :+1:

**Comment by friendyou1** [comment]
I'm the Electronic person to reserve in a sad moment.

**Review by friendyou1 — COMMENTED** [review]
Not worthy @Fylax

**Review by Fylax — APPROVED** [review]

**Comment by graciousgrey** [comment]
Is this one ready to be merged ? :)

**Comment by haimkastner** [comment]
Hi @graciousgrey , from my side, yes.

**Comment by graciousgrey** [comment]
Thank you :)

**Pull request merged**